test/opt/graphics_robust_access_test.cpp

```diff
@@ -0,0 +1,1307 @@
+    SinglePassRunAndMatch<GraphicsRobustAccessPass>(shaders.str(), true);
+  }
+}
+
+TEST_F(GraphicsRobustAccessTest, ACVectorExcessConstantClamped) {
+  for (auto* ac : AccessChains()) {
+    std::ostringstream shaders;
+    shaders << ShaderPreambleAC() << TypesVoid() << TypesInt() << R"(
+       %v4uint = OpTypeVector %uint 4
+       %var_ty = OpTypePointer Function %v4uint
+       %ptr_ty = OpTypePointer Function %uint
+       %uint_4 = OpConstant %uint 4
+       )"
+            << MainPrefix() << R"(
+       %var = OpVariable %var_ty Function)" << ACCheck(ac, "%uint_4", "%uint_3")
+            << MainSuffix();
+    SinglePassRunAndMatch<GraphicsRobustAccessPass>(shaders.str(), true);
+  }
+}
+
+TEST_F(GraphicsRobustAccessTest, ACVectorNegativeConstantClamped) {
+  for (auto* ac : AccessChains()) {
+    std::ostringstream shaders;
+    shaders << ShaderPreambleAC() << TypesVoid() << TypesInt() << R"(
+       %v4uint = OpTypeVector %uint 4
+       %var_ty = OpTypePointer Function %v4uint
+       %ptr_ty = OpTypePointer Function %uint
+       %int_n1 = OpConstant %int -1
+       )"
+            << MainPrefix() << R"(
+       ; CHECK: %int_0 = OpConstant %int 0
+       %var = OpVariable %var_ty Function)" << ACCheck(ac, "%int_n1", "%int_0")
+            << MainSuffix();
+    SinglePassRunAndMatch<GraphicsRobustAccessPass>(shaders.str(), true);
+  }
+}
+
+// Like the previous test, but ensures the pass knows how to modify an index
+// which does not come first in the access chain.
+TEST_F(GraphicsRobustAccessTest, ACVectorInArrayNegativeConstantClamped) {
+  for (auto* ac : AccessChains()) {
+    std::ostringstream shaders;
+    shaders << ShaderPreambleAC() << TypesVoid() << TypesInt() << R"(
+       %v4uint = OpTypeVector %uint 4
+       %uint_1 = OpConstant %uint 1
+       %uint_2 = OpConstant %uint 2
+       %arr = OpTypeArray %v4uint %uint_2
+       %var_ty = OpTypePointer Function %arr
+       %ptr_ty = OpTypePointer Function %uint
+       %int_n1 = OpConstant %int -1
+       )"
+            << MainPrefix() << R"(
+       ; CHECK: %int_0 = OpConstant %int 0
+       %var = OpVariable %var_ty Function)"
+            << ACCheck(ac, "%uint_1 %int_n1", "%uint_1 %int_0") << MainSuffix();
+    SinglePassRunAndMatch<GraphicsRobustAccessPass>(shaders.str(), true);
+  }
+}
+
+TEST_F(GraphicsRobustAccessTest, ACVectorGeneralClamped) {
+  for (auto* ac : AccessChains()) {
+    std::ostringstream shaders;
+    shaders << ShaderPreambleAC({"i"}) << TypesVoid() << TypesInt() << R"(
+       %v4uint = OpTypeVector %uint 4
+       %var_ty = OpTypePointer Function %v4uint
+       %ptr_ty = OpTypePointer Function %uint
+       %i = OpUndef %int)"
+            << MainPrefix() << R"(
+       ; CHECK: %[[GLSLSTD450:\w+]] = OpExtInstImport "GLSL.std.450"
+       ; CHECK-DAG: %int_0 = OpConstant %int 0
+       ; CHECK-DAG: %int_3 = OpConstant %int 3
+       ; CHECK: OpLabel
+       ; CHECK: %[[clamp:\w+]] = OpExtInst %int %[[GLSLSTD450]] UClamp %int_0 %i %int_3
```

Comment:
Shouldn't this be UClamp %i %int_0 %int_3?

Reply:
Yes.  Fixed in next commit